source/MaterialXTest/ShaderValid.cpp

```diff
@@ -367,192 +365,68 @@ static void runValidation(const std::string& shaderName, std::vector<mx::Element
-          </nodegraph> \
-        </materialx>";
-
-        MaterialX::readFromXmlBuffer(doc, lightDoc.c_str());
-        mx::NodeGraphPtr nodeGraph = doc->getNodeGraph("lighting1");
-        mx::OutputPtr output = nodeGraph->getOutput("lighting_output");
-
-        // Run validation
-        std::vector<mx::ElementPtr> outputList;
-        outputList.push_back(output);
-        runValidation(nodeGraph->getName(), outputList, validator, shaderGenerator, orthographicView, doc, log);
-    }
-
-    //
-    // Materials test
-    //
-    {
-        std::vector<mx::MaterialPtr> materials;
-        createExampleMaterials(doc, materials);
-
-        for (const mx::MaterialPtr& material : materials)
-        {
-            for (mx::ShaderRefPtr shaderRef : material->getShaderRefs())
-            {
-                const std::string name = material->getName() + "_" + shaderRef->getName();
-
-                std::vector<mx::ElementPtr> shaderRefList;
-                shaderRefList.push_back(shaderRef);
-                runValidation(name, shaderRefList, validator, shaderGenerator, orthographicView, doc, log);
-            }
-        }
-    }
-
-
+  #ifdef LOG_TO_FILE
+      std::ofstream logfile("log_shadervalid_glsl_materialx_documents.txt");
+      std::ostream& log(logfile);
+  #else
+      std::ostream& log(std::cout);
+  #endif
+
+      // Library search path
+      mx::FilePath searchPath = mx::FilePath::getCurrentPath() / mx::FilePath("documents/Libraries");
+
+      // Create a validator
+      bool orthographicView = true;
+      mx::GlslValidatorPtr validator = createValidator(orthographicView, "", log);
+
+      // Set up shader generator
+      mx::ShaderGeneratorPtr shaderGenerator = mx::GlslShaderGenerator::create();
+      shaderGenerator->registerSourceCodeSearchPath(searchPath);
+
+      const std::string path = "documents/Tests";
+      mx::StringVec files;
+      mx::getDocumentsInDirectory(path, files);
+      for (std::string file : files)
+      {
+          std::string filename = path + "/" + file;
+          if (endsWithCaseInsensitive(filename, ".mtlx"))
+          {
+              log << "MTLX Filename: " << filename << std::endl;
+              mx::DocumentPtr doc = mx::createDocument();
+              readFromXmlFile(doc, filename);
+
+              std::vector<mx::NodeGraphPtr> nodeGraphs = doc->getNodeGraphs();
+              std::vector<mx::OutputPtr> outputList;
+              for (mx::NodeGraphPtr nodeGraph : nodeGraphs)
+              {
+                  log << "NodeGraph: " << nodeGraph->getName() << std::endl;
+                  std::vector<mx::OutputPtr> nodeGraphOutputs = nodeGraph->getOutputs();
+                  for (mx::OutputPtr output : nodeGraphOutputs)
+                  {
+                      outputList.push_back(output);
+                  }
+              }
+              loadLibraries({ "stdlib", "sxpbrlib" }, searchPath, doc);
```

Comment:
Assume we'll fix to use importLibrary later.

Reply:
Yep